File: lib/python/benchmarks.py
import timeit
from db import db

dba = db("localhost", 3434, "admin", "password")

def run():
    for i in range(1000):
        path = "/" + str(i)
        print(dba.push(path, "HELLO WORLD!"))
        print(dba.get(path))
        print(dba.delete(path))

print(timeit.timeit("run()", setup="from __main__ import run", number=1))

"""
BENCHMARKS:

3.047855800999969
3.5085566739999194
"""

File: lib/ruby/benchmarks.rb
require 'benchmark'
require_relative "db"

dab = DB.new "localhost", 3434, "admin", "password"

def puts(string)
    #do not output to console
    #just take performance
end

def test(db)
    for i in (0..1000).to_a do
        path = "/" + i.to_s
        puts db.push(path, "HELLO WORLD")
        puts db.get(path)
        puts db.delete(path)
    end
end


def no(db)
    for i in (0..1000).to_a do
        path = "/" + i.to_s
        db.push(path, "HELLO WORLD")
        db.get(path)
        db.delete(path)
    end
end

def re(db)
    for i in (0..1000).to_a do
        path = "/" + i.to_s
        db.push(path, "HELLO WORLD")
        puts db.get(path)
        db.delete(path)
    end
end

n = 50000
Benchmark.bm(1) do |x|
  x.report("write/read/gets with puts:")   { test(dab) }
  x.report("write/read/gets with no puts:") { no(dab) }
  x.report("write/read/gets with puts on DB.get:") { re(dab) }
end

=begin
with puts
=end